compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -760,7 +760,9 @@ impl<'f, 'tcx> Coerce<'f, 'tcx> {
         {
             if a_data.principal_def_id() == b_data.principal_def_id() {
                 return self.unify_and(a, b, |_| vec![]);
-            } else if !self.tcx().features().trait_upcasting {
+            }
+            // FIXME(dyn-star): `dyn*` upcasting is currently broken and ICEy.
+            /* else if !self.tcx().features().trait_upcasting {
```

Comment:
I'm uncomfortable leaving this commented out.

Reply:
Are you asking for it to be deleted instead? Because I can do that.